quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs

```diff
@@ -54,6 +54,10 @@ pub struct DefaultDocMapperBuilder {
     /// Name of the fields that are tagged.
     #[serde(default)]
     pub tag_fields: Vec<String>,
+    /// The partition key is a DSL used to route documents
+    /// into specific splits.
+    #[serde(default)]
+    pub partition_key: String,
```

Comment:
Can't we use an `Option<String>` just like we already do for `timestamp_field` or `demux_field`?

Reply:
These are very different. 
Here the empty string is not expression the absence of a partitioning key. It is describing a partitionin key that is the empty string "".